core/chains/evm/txmgr/client.go

```diff
@@ -0,0 +1,154 @@
+	return &evmTxmClient{client: c}
+}
+
+func (c *evmTxmClient) BatchSendTransactions(ctx context.Context, txStore EvmTxStore, attempts []EvmTxAttempt, batchSize int, lggr logger.Logger) (codes []clienttypes.SendTxReturnCode, txErrs []error, err error) {
+	// preallocate
+	codes = make([]clienttypes.SendTxReturnCode, len(attempts))
+	txErrs = make([]error, len(attempts))
+
+	reqs, batchErr := batchSendTransactions(ctx, txStore, attempts, batchSize, lggr, c.client)
+	err = errors.Join(err, batchErr) // this error does not block processing
+
+	// safety check - exits before processing
+	if len(reqs) != len(attempts) {
+		lenErr := fmt.Errorf("Returned request data length (%d) != number of tx attempts (%d)", len(reqs), len(attempts))
+		err = errors.Join(err, lenErr)
+		lggr.Criticalw("Mismatched length", "error", err)
+		return
+	}
+
+	// for each batched tx convert response to standard error code
+	for i := range reqs {
+		// convert to tx for logging purposes - exits early if error occurs
+		tx, signedErr := GetGethSignedTx(attempts[i].SignedRawTx)
+		if signedErr != nil {
+			err = errors.Join(err, signedErr)
+			return
+		}
+		codes[i], txErrs[i] = evmclient.NewSendErrorReturnCode(reqs[i].Error, lggr, tx, attempts[i].Tx.FromAddress, c.client.IsL2())
+	}
+	return
+}
+
+func (c *evmTxmClient) SendTransactionReturnCode(ctx context.Context, etx EvmTx, attempt EvmTxAttempt, lggr logger.Logger) (clienttypes.SendTxReturnCode, error) {
+	signedTx, err := GetGethSignedTx(attempt.SignedRawTx)
+	if err != nil {
+		lggr.Criticalw("Fatal error signing transaction", "err", err, "etx", etx)
+		return clienttypes.Fatal, err
+	}
+	return c.client.SendTransactionReturnCode(ctx, signedTx, etx.FromAddress)
+}
+
+func (c *evmTxmClient) PendingNonceAt(ctx context.Context, fromAddress common.Address) (n int64, err error) {
+	nextNonce, err := c.client.PendingNonceAt(ctx, fromAddress)
+	if err != nil {
+		return n, err
+	}
+
+	if nextNonce > math.MaxInt64 {
+		return n, fmt.Errorf("nonce overflow, got: %v", nextNonce)
+	}
+	return int64(nextNonce), nil
+}
+
+func (c *evmTxmClient) SequenceAt(ctx context.Context, addr common.Address, blockNum *big.Int) (evmtypes.Nonce, error) {
+	return c.client.SequenceAt(ctx, addr, blockNum)
+}
+
+func (c *evmTxmClient) BatchGetReceipts(ctx context.Context, attempts []EvmTxAttempt) (txReceipt []*evmtypes.Receipt, txErr []error, funcErr error) {
+	var reqs []rpc.BatchElem
+	for _, attempt := range attempts {
+		req := rpc.BatchElem{
+			Method: "eth_getTransactionReceipt",
+			Args:   []interface{}{attempt.Hash},
+			Result: &evmtypes.Receipt{},
+		}
+		reqs = append(reqs, req)
+	}
+
+	if err := c.client.BatchCallContext(ctx, reqs); err != nil {
+		return nil, nil, fmt.Errorf("EthConfirmer#batchFetchReceipts error fetching receipts with BatchCallContext: %w", err)
+	}
+
+	for _, req := range reqs {
+		result, err := req.Result, req.Error
+
+		receipt, ok := result.(*evmtypes.Receipt)
```

Comment:
nit/ you could avoid the casting logic by loading up `txReceipt` with `&evmtypes.Receipt{}`s, and then passing those pointers along to `Result: txReceipt[i]`.

Reply:
sounds good to me! I will address it in a later PR - https://smartcontract-it.atlassian.net/browse/BCI-1319